src/grappler/ngraph_optimizer.cc

```diff
@@ -108,7 +108,7 @@ Status NgraphOptimizer::Optimize(tensorflow::grappler::Cluster* cluster,
   std::set<string> nodes_to_preserve;
 
   // Feed Nodes
-  for (int i = 0; i < item.feed.size(); i++) {
+  for (unsigned int i = 0; i < item.feed.size(); i++) {
```

Comment:
Change this to auto or size_t. Also do not use c style cast- always use C++ case i.e., siatic_cast<> etc. when needed.